src/transport/CASESession.cpp

```diff
@@ -217,47 +572,353 @@ CHIP_ERROR CASESession::HandleSigmaR2_and_SendSigmaR3(const PacketHeader & heade
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
 
-    uint16_t data_len = 0;
+    err = HandleSigmaR2(header, msg);
+    SuccessOrExit(err);
 
-    const uint8_t * buf = msg->Start();
+    err = SendSigmaR3();
+    SuccessOrExit(err);
 
-    System::PacketBufferHandle msg_R3;
+exit:
+    return err;
+}
+
+CHIP_ERROR CASESession::HandleSigmaR2(const PacketHeader & header, const System::PacketBufferHandle & msg)
+{
```

Comment:
This is a 200-line method with 15 local variables and 2 arguments. Could we split it into something more manegeable?
Could we use some RAII to avoid the need for goto exit and then not be forced to declare all variables at top (somewhat better scoping to help readability)?